src/Http/Controllers/VoyagerBaseController.php

```diff
@@ -466,6 +467,138 @@ public function restore(Request $request, $id)
         return redirect()->route("voyager.{$dataType->slug}.index")->with($data);
     }
 
+    //***************************************
+    //
+    //  Delete uploaded file
+    //
+    //****************************************
+
+    public function remove_media(Request $request)
+    {
+        try {
+            // GET THE SLUG, ex. 'posts', 'pages', etc.
+            $slug = $request->get('slug');
+
+            // GET file name
+            $filename = $request->get('filename');
+
+            // GET record id
+            $id = $request->get('id');
+
+            // GET field name
+            $field = $request->get('field');
+
+            // GET multi value
+            $multi = $request->get('multi');
+
+            $dataType = Voyager::model('DataType')->where('slug', '=', $slug)->first();
+
+            // Load model and find record
+            $model = app($dataType->model_name);
+            $data = $model::find([$id])->first();
+
+            // Check if field exists
+            if (!isset($data->{$field})) {
+                throw new Exception(__('voyager::generic.field_does_not_exist'), 400);
+            }
+
+            // Check permission
+            $this->authorize('edit', $data);
```

Comment:
Why is this `edit`?

Reply:
Because the only place where we can delete uploaded files is when we are changing BREAD data and the only permission the user needs is edit.

Reply:
But you can delete things also when adding right?

Reply:
To be honest I'm not 100% sure but untill you are not done with the add part there shouldn't be anything to delete either in database or in filesystem.
I'll check tomorrow anyway I'm not in front of a PC right now.

Reply:
I can confirm the delete button will only appear on edit and there wouldn't be anything to delete on add.